pygeos/strtree.py

```diff
@@ -32,15 +48,41 @@ def __init__(self, geometries, leafsize=5):
     def __len__(self):
         return self._tree.count
 
-    def query(self, envelope):
-        """Return all items whose extent intersect the given envelope.
+    def query(self, geometry, predicate=None):
+        """Return all items whose extent intersect the envelope of the input
+        geometry.  If predicate is provided, these items are limited to those
+        that satisfy the predicate operation when compared against the input
+        geometry.
+
+        If geometry is None, an empty array is returned.
 
         Parameters
         ----------
-        envelope : Geometry
-            The envelope of the geometry is taken automatically.
+        geometry : Geometry
+            The envelope of the geometry is taken automatically for
+            querying the tree.
+        predicate : str, optional (default: None)
+            The predicate to use for testing geometries from the tree
+            that are within the input geometry's envelope.
```

Comment:
I think we should clarify here in which order the predicate is evaluated (predicate(passed_geometry, tree_geometry) vs predicate(tree_geometry, passed_geometry))

(it's not directly self-evident which of the two it would be I think)

Reply:
I tried to clarify this, and also added an example to the docstring.